### PR TITLE
core: add pin_defs for MK66F18

### PR DESCRIPTION
**Pull request by stapelberg**
## Description

This is in preparation for https://github.com/qmk/qmk_firmware/pull/10171

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: tmk_core/common/chibios/pin_defs.h [hunk 1]
```diff
@@ -53,6 +53,183 @@
 #        define D5 PAL_LINE(GPIOC, 13)
 #        define B0 PAL_LINE(GPIOC, 14)
 #    endif
+#elif defined(MK66F18)
```
Comment (zvecr) [inline tmk_core/common/chibios/pin_defs.h:56]: This whole block is not required, and the extra values > 15 should just be added to the non ctpc branch

**Review by zvecr — CHANGES_REQUESTED** [review]

File: tmk_core/common/chibios/pin_defs.h (hunk 1 above)
Comment (stapelberg) [reply tmk_core/common/chibios/pin_defs.h]: Done.

**Review by fauxpark — APPROVED** [review]

**Comment by stapelberg** [comment]
Looks like travis timed out: The job exceeded the maximum time limit for jobs, and has been terminated.

Could one of you restart it please?

**Comment by fauxpark** [comment]
Restarting won't help.

**Comment by stapelberg** [comment]
> Restarting won't help.

Okay. I assume the Travis breakage is unrelated, though.

What’s the next step in getting this merged?

I think I accidentally removed the review request from qmk/collaborators as a side effect of re-requesting a review via the GitHub UI. Just want to make sure that doesn’t accidentally mess up the workflow.

**Review by tzarc — APPROVED** [review]

**Comment by tzarc** [comment]
> What’s the next step in getting this merged?

Not much!

**Pull request merged**